src/Type/Php/StrContainingTypeSpecifyingExtension.php

```diff
@@ -77,6 +82,15 @@ public function specifyTypes(FunctionReflection $functionReflection, FuncCall $n
 					$context,
 					false,
 					$scope,
+					new BooleanAnd(
+						new NotIdentical(
+							$args[$needleArg]->value,
+							new String_(''),
+						),
+						new FuncCall(new Name('FAUX_FUNCTION'), [
```

Comment:
thx for getting this over the finishing line <3.

could you elaborate a bit, what this FAUX_FUNCTION thing is about?
is this a kind-of workaround, which should be encapsulated in a helper somehow, so the code gets readable for non-phpstan-hard-core users :-)?

Reply:
it's kind of the soft force flag xD

Reply:
on the other hand this is veeery special and the expressions is different for each case maybe

Reply:
The thing that's problematic for `str_contains()` type-specifying extensions and similar is that PHPStan thinks the function is `is_string_non_empty()` because it narrows down the type to `non-empty-string`. But there are other properties of the narrowed string than that - in this case that it contains another string which can't be expressed by the PHPStan typesystem.

So when you call `str_contains($nonEmptyString)` PHPStan thinks it's `is_string_non_empty($nonEmptyString)` and that it has to be always true. But in fact it should be `is_string_non_empty($nonEmptyString) && string_contains_string(...)`. So that's what I'm simulating here.

I don't think we need a helper here.